design/baremetal-operator/secure-boot.md

```diff
@@ -0,0 +1,101 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Secure boot support
+
+## Status
+
+implementable
+
+## Summary
+
+This design proposes exposing an ability to turn UEFI secure boot on and off
+during provisioning and deprovisioning.
+
+## Motivation
+
+Security-conscious deployments would like to make sure secure boot is enabled
+for their instances, so that the hardware refuses to boot kernel-level code
+that has not been signed with a known key.
+
+### Goals
+
+- API addition to enable secure boot before booting the instance (and disable
+  it on deprovisioning)
+
+### Non-Goals
+
+- Support for custom secure boot keys.
+- Secure boot during deployment/cleaning/inspection.
+
+## Proposal
+
+## Design Details
+
+Add a new value for `BootMode` enumeration: `UEFI secure boot` (constant name
```

Comment:
The Enum value needs updating to reflect the discussion on the implementation PR